src/core/sensor/qgsiodevicesensor.cpp

```diff
@@ -0,0 +1,338 @@
+/***************************************************************************
+                             qgsiodevicesensor.cpp
+                             ---------------------------
+    begin                : March 2023
+    copyright            : (C) 2023 by Mathieu Pellerin
+    email                : mathieu at opengis dot ch
+ ***************************************************************************/
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsiodevicesensor.h"
+#include "qgssensorregistry.h"
+
+#include <QDomElement>
+
+#if defined( Q_OS_ANDROID ) || defined( Q_OS_LINUX )
+#include <sys/socket.h>
+#endif
+
+QgsIODeviceSensor::~QgsIODeviceSensor()
+{
+}
+
+void QgsIODeviceSensor::initIODevice( QIODevice *device )
+{
+  mIODevice.reset( device );
+
+  if ( mIODevice )
+  {
+    connect( mIODevice.get(), &QIODevice::readyRead, this, &QgsIODeviceSensor::parseData );
+  }
+}
+
+QIODevice *QgsIODeviceSensor::iODevice() const
+{
+  return mIODevice.get();
+}
+
+void QgsIODeviceSensor::parseData()
+{
+  QgsAbstractSensor::SensorData data;
+  data.lastValue = mIODevice->readAll();
+  data.lastTimestamp = QDateTime::currentDateTime();
+  setData( data );
+}
+
+//--------------
+
+QgsTcpSocketSensor::QgsTcpSocketSensor( QObject *parent )
+  : QgsIODeviceSensor( parent )
+  , mTcpSocket( new QTcpSocket() )
+{
+  connect( mTcpSocket, &QAbstractSocket::stateChanged, this, &QgsTcpSocketSensor::socketStateChanged );
+  initIODevice( mTcpSocket );
+}
+
+QgsTcpSocketSensor *QgsTcpSocketSensor::create( QObject *parent )
+{
+  return new QgsTcpSocketSensor( parent );
+}
+
+QString QgsTcpSocketSensor::type() const
+{
+  return QLatin1String( "tcp_socket" );
+}
+
+QString QgsTcpSocketSensor::hostName() const
+{
+  return mHostName;
+}
+
+void QgsTcpSocketSensor::setHostName( const QString &hostName )
+{
+  if ( mHostName == hostName )
+    return;
+
+  mHostName = hostName;
+}
+
+int QgsTcpSocketSensor::port() const
+{
+  return mPort;
+}
+
+void QgsTcpSocketSensor::setPort( int port )
+{
+  if ( mPort == port || port < 1 )
```

Comment:
Shouldn't you also check if `port > 65535`? (also in udp code)

Reply:
@uclaros , good point; I'll adjust in the other open PR as to avoid another 2 hours or CI (:sob:).  Thanks for having had a look.